doc/rst/technotes/extern.rst

```diff
@@ -153,7 +157,22 @@ The c_ptr(T) type is a generic type representing a C pointer to an arbitrary
 type T. This pointer should normally only point to local memory - since no
 communication will be generated when it is dereferenced.  Of course, the
 pointed-to type T should be one that is supported in C interoperability if the
-c_ptr(T) is used for C interoperability.
+c_ptr(T) is used for C interoperability. The c_ptr(T) type supports
+indexing to get a reference to the i'th element.
+
+c_array(T,n)
+~~~~~~~~~~~~
+
+The c_array(T,n) type is a generic value type representing a C fixed-size
+array. Here n must be known at compile-time.
```

Comment:
I think it would look better if the letter n was in quotes or something:
'n'
``n``

Up to you.